spec/grover/middleware_spec.rb

```diff
@@ -860,6 +860,46 @@
       end
     end
 
+    describe '#ignore_request?' do
+      context 'when app configuration has a Proc for ignore_request' do
+        context 'with a hostname' do
+          before { allow(Grover.configuration).to receive(:ignore_request).and_return(->(req) { req.host == 'www.example.org' }) }
+
+          it 'request is ignored when the request path passed to the proc defined in ignore_request returns true' do
+            response = get 'http://www.example.org/foobazbar'
+            expect(response.content_type).to eq 'text/html'
+
+            response = get 'http://www.example.org/foobazbar.pdf'
+            expect(response.content_type).to eq 'text/html'
+
+            response = get 'http://www.example.org/foobarbar'
+            expect(response.content_type).to eq 'text/html'
+
+            response = get 'http://www.therealexample.org/foobarbar.pdf'
+            expect(response.content_type).to eq 'application/pdf'
+          end
+        end
+
+        context 'with a custom header' do
+          before { allow(Grover.configuration).to receive(:ignore_request).and_return(->(req) { req.has_header?('X-BLOCK') }) }
+
+          it 'request is ignored when the request path passed to the proc defined in ignore_request returns true' do
+            response = get 'http://www.example.org/foobazbar', {}, { 'X-BLOCK' => '1' }
+            expect(response.content_type).to eq 'text/html'
+
+            response = get 'http://www.example.org/foobazbar.pdf', {}, { 'X-BLOCK' => '1' }
+            expect(response.content_type).to eq 'text/html'
+
+            response = get 'http://www.example.org/foobarbar', {}, { 'X-BLOCK' => '1' }
```

Comment:
not sure what this is adding to the tests? Ideally this and the next test would both NOT have the extra header. And I would suggest that the URI requested be the same for all (except for the `.pdf` extension where appropriate!)